### PR TITLE
update index image build ci

### PR DESCRIPTION
**Pull request by pindge**
fixes: #91 

## Scope
- update readme for creating new index image
- update github build ci to allow unstable/dev index image build before tagging image with new number

### CODE REVIEW

File: index/readme.md [hunk 1]
```diff
@@ -1,17 +1,55 @@
-# opendatacube/datacube-index
 
+# opendatacube/datacube-index
 Dockerfile for use in indexing into the Open Data Cube
 
+The build images are hosted here: https://hub.docker.com/r/opendatacube/datacube-index.
```
Comment (jmettes) [inline index/readme.md:5]: Would it be worth also linking back to the github repo? That way there's a way to find it from DockerHub

Comment (pindge) [reply index/readme.md:5]: yes

**Comment by jmettes** [comment]
I'm a bit confused on the distinction of dev tags and releasing

**Comment by pindge** [comment]
> 

There isn't a written guideline that opendatacube follows. from observation for a few active developed repos,

https://hub.docker.com/r/opendatacube/explorer/tags
https://hub.docker.com/r/opendatacube/ows/tags
https://hub.docker.com/r/opendatacube/datacube-alchemist/tags
... list goes on

the tag is `base tagged` if tagged properly, usually via github release number + release notes, then each non tagged pr merged will just build a `dev` latest, so you can see the image numbers are `base`, the next one is `base+dev`, until the next incremental released tag.

Our issue is this datacube-docker repo does not do tag release, which to me feels more like the whole idea of this repo setup is problematic.

```
(odc-tests-py38) ubuntu@:~/datacube-docker$ git tag
(odc-tests-py38) ubuntu@:~/datacube-docker$ cd ~/datacube-ows
(odc-tests-py38) ubuntu@:~/datacube-ows$ git tag
0.10.0
0.11.0
```

**Review by emmaai — APPROVED** [review]

**Pull request merged**